### ISSUE TITLE
EmptyLineSeparator check does not validate newlines before class and after last method

### ISSUE THREAD

**Issue by mkordas** [checkstyle/checkstyle#2067]
The following input should cause two violations:
```java
import java.util.Map;


class InputLineEmptiness { // two empy lines above
    void a() {
    } // two empty lines below


}
```
when the config is set to:
```xml
<module name="EmptyLineSeparator">
  <property name="allowMultipleEmptyLines" value="false"/>
</module>
```
but it doesn't raise any violation.

Generally this check validates if there is no more than one empty line _above_ some tokens. It works well when there are multiple unnecessary spaces above `import` or `package`. When the check reaches `class` declaration it also would validate lines above, but...

 - at line [EmptyLineSeparatorCheck.java#L214][1] next sibling is obtained for `class` token and it is `null`, as there's noting else on the same level as this class in the file
 - because next sibling is `null` entire validation is skipped at line [EmptyLineSeparatorCheck.java#L216][2]

  [1]: https://github.com/checkstyle/checkstyle/blob/checkstyle-6.10.1/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java#L214
  [2]: https://github.com/checkstyle/checkstyle/blob/checkstyle-6.10.1/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java#L216

The check also doesn't validate newlines after last method. To do that, check probably should also look _below_ elements, not only _above_.

**Comment by Vladlis** [comment]
@mkordas 
> The check also doesn't validate newlines after last method. To do that, check probably should also look below elements, not only above.

Please, add more details as for now this functionality is missed at all:
- should the check validate newlines only after the last method or after whatever last token in frame? (from specified in config tokens)?
- should this validation happen only when ```allowMultipleEmptyLines``` is false? or always? or a new option is needed?
- should we check for any newlines or only for multiple (more than one)?
- anything else?

**Comment by mkordas** [comment]
> should the check validate newlines only after the last method or after whatever last token in frame? (from specified in config tokens)?

ideally all tokens, but if it's not easy we can move this to separate issue and do just methods here

> should this validation happen only when allowMultipleEmptyLines is false? or always? or a new option is needed?

yes, only when `allowMultipleEmptyLines` is `false`

> should we check for any newlines or only for multiple (more than one)
 
0 lines is ok, 1 line is ok, more than 1 line is not ok when `allowMultipleEmptyLines` is `false`

> anything else

we can raise separate issue if something is missing here

**Comment by romani** [comment]
merged.

**Comment by vovkab** [comment]
I think it still not working correctly everywhere, for example:

```
    public static final ....;






    /**
    */
```

this will pass

**Comment by romani** [comment]
Please create new issue on this
